src/aind_codeocean_api/credentials.py

```diff
@@ -1,46 +1,137 @@
 """Basic CodeOcean Credentials Handling."""
+import functools
 import json
 import os
+from abc import ABC, abstractmethod
 from pathlib import Path
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Tuple, Type, Union
 
 import boto3
-from pydantic import BaseSettings, Field, SecretStr, validator
-from pydantic.env_settings import (
+from pydantic import Field, SecretStr, field_validator
+from pydantic.fields import FieldInfo
+from pydantic_settings import (
+    BaseSettings,
     EnvSettingsSource,
     InitSettingsSource,
-    SecretsSettingsSource,
+    PydanticBaseSettingsSource,
+    SettingsConfigDict,
 )
 
 
-# Small helper function to get an aws secret
-def get_secret(secret_name: str) -> dict:
-    """
-    Retrieves a secret from AWS Secrets Manager.
+class JsonConfigSettingsSource(PydanticBaseSettingsSource, ABC):
+    """Abstract base class for settings that parse json"""
 
-    param secret_name: The name of the secret to retrieve.
-    """
-    # Create a Secrets Manager client
-    client = boto3.client("secretsmanager")
-    try:
-        response = client.get_secret_value(SecretId=secret_name)
-    finally:
-        client.close()
-    return json.loads(response["SecretString"])
+    def __init__(self, settings_cls, config_file_location):
+        """
+        Class constructor for generic settings source that parses json
+        Parameters
+        ----------
+        settings_cls
+          Required for parent init
+        config_file_location
+          Location of json contents to parse
+        """
+        self.config_file_location = config_file_location
+        super().__init__(settings_cls)
+
+    @abstractmethod
+    def _retrieve_contents(self) -> Dict[str, Any]:
+        """Retrieve contents from config_file_location"""
+
+    @functools.cached_property
+    def _json_contents(self):
+        """Cache contents to a property to avoid re-downloading."""
+        contents = self._retrieve_contents()
+        return contents
+
+    def get_field_value(
+        self, field: FieldInfo, field_name: str
+    ) -> Tuple[Any, str, bool]:
+        """This function needs to be implemented for every
+        PydanticBaseSettingsSource"""
+        file_content_json = self._json_contents
+        field_value = file_content_json.get(field_name)
+        return field_value, field_name, False
+
+    def prepare_field_value(
```

Comment:
What is the purpose of having this method if it only returns the value?

Reply:
I followed this example from the pydantic documentation directly. I think these methods need to be implemented.